lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -152,12 +153,20 @@ export class ResponseHandler {
                 this.clientId,
                 idTokenObj.claims.tid
             );
+
+            cachedAccount = this.generateAccountEntity(
+                serverTokenResponse,
+                idTokenObj,
+                authority
+            );
         }
 
         // AccessToken
         let cachedAccessToken: AccessTokenEntity = null;
         if (!StringUtils.isEmpty(serverTokenResponse.access_token)) {
-            const responseScopes = ScopeSet.fromString(serverTokenResponse.scope);
+            
+            // If scopes not returned in server response, use request scopes
+            const responseScopes = serverTokenResponse.scope ? ScopeSet.fromString(serverTokenResponse.scope) : new ScopeSet(requestScopes || []);
```

Comment:
What is the scenario where the server wouldn't return scopes on an access token?

Reply:
The client credentials flow does not return scopes in the response. I believe MSAL Java has a similar handling pattern for this.